Change standardRequests hover label

Seba244c opened the pull request:
Closes https://github.com/ldtteam/minecolonies-features/issues/315

# Changes proposed in this pull request:
- Changed standard request list UI, hover text to be more human-readable
Change the hover text in the request lists such as in the postbox to be more human-readable. RN when a request is first in the queue it says "#0 in queue", which does not make sense. But in this commit, it adds 1 to that value. So when something is first in the request queue it says "#1 in queue".

Review please

Seba244c committed:
Change standardrquests in queue label

Seba244c commented:
![Jair Coby](https://user-images.githubusercontent.com/23051740/110476535-5f9cae00-80e2-11eb-94a6-9fed4a38d870.PNG)

Raycoms commented:
If we go so far, at 0 it should say "In progress" and for the others "#x in queue"

Seba244c commented:
Good idea let me try to implement that.

Seba244c commented:
Just found out I placed the `+1`, in the wrong place 🤦‍♂️. Fixing it now and testing it out with the new functionality you mentioned.

Seba244c committed:
Changed hover text when first in queue

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequests.java:
```diff
@@ -275,7 +275,8 @@ public ITextComponent getShortDisplayString()
             if (posInList >= 0)
             {
-                return ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_QUEUE+1, posInList));
+            	if(posInList == 0) 	return ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_PROGRESS, posInList+1));
```
in line return posInList == 0 ? ... : ...

Raycoms replied:
in progress doesn't need the pos + 1 param

Seba244c replied:
Oh yea, sry forgot to remove that part, when copying.

Seba244c committed:
Fix inputing args when args wasnt used

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequests.java:
```diff
@@ -275,7 +275,8 @@ public ITextComponent getShortDisplayString()
             if (posInList >= 0)
             {
-                return ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_QUEUE, posInList));
+            	if(posInList == 0) 	return ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_PROGRESS));
```
Still inline it with ? :

Seba244c replied:
WDYM?

Raycoms replied:
return posInList == 0 ? ImmutableList.of() : ImmutableList.of()

Seba244c replied:
Oh `?:` have only slightly heard of it, and never used it. Gimme a sec

Seba244c replied:
Thank you for showing me this

Raycoms replied:
Np

Seba244c committed:
Make return if statements one line

Raycoms commented:
Final comment, do we need the +1 still, isn't it fine to have = 0 => in progress, and the others are then in the queue 1,2,3,4 following that?

Seba244c committed:
Removed +1 since so the first in queue wouldnt show as the second

Seba244c commented:
Oh, yea thought on that, just forgot since Eclipse decided to crash on me. Good idea and I am sry for the mess I have created with this PR, it's my first PR ever

Raycoms commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequests.java:
```diff
@@ -399,7 +399,7 @@ public final ITextComponent getShortDisplayString()
                 if (posInList >= 0)
                 {
-                    return ImmutableList.of(new TranslationTextComponent(AT, requester), new TranslationTextComponent(IN_QUEUE, posInList));
+                	return posInList == 0 ? ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_PROGRESS)) : ImmutableList.of(new TranslationTextComponent(FROM, requester), new TranslationTextComponent(IN_QUEUE, posInList));
```
Here it's AT not FROM (last nitpick, I swear!)

Seba244c replied:
Wow another Copypasta error

Raycoms commented:
> Oh, yea thought on that, just forgot since Eclipse decided to crash on me. Good idea and I am sry for the mess I have created with this PR, it's my first PR ever

No problem, it's always a bit awkward entering a new project like that

Seba244c committed:
Fixed another copy pasta error

ravenbuilder934 commented on src/main/resources/assets/minecolonies/lang/en_us.json:
```diff
@@ -1734,6 +1734,7 @@
   "com.minecolonies.coremod.from": "From: %s",
   "com.minecolonies.coremod.at": "At: %s",
   "com.minecolonies.coremod.listentry": "#%d in queue",
+  "com.minecolonies.coremod.in_progrss": "In Progress",
```
```suggestion
  "com.minecolonies.coremod.in_progress": "In Progress",
```

Seba244c replied:
Fixed

ravenbuilder934 commented on src/api/java/com/minecolonies/api/util/constant/TranslationConstants.java:
```diff
@@ -438,6 +438,8 @@
     @NonNls
     public static final String IN_QUEUE                                                             = "com.minecolonies.coremod.listentry";
     @NonNls
+    public static final String IN_PROGRESS                                                          = "com.minecolonies.coremod.in_progrss";
```
```suggestion
    public static final String IN_PROGRESS                                                          = "com.minecolonies.coremod.in_progress";
```

ravenbuilder934 reviewed (changes requested):
Have you tested this? Can you provide screenshots?

Seba244c commented:
No, I haven't tested this. I do not have a 1.16.5 colony(usually plays in 1.15), just created a new creative 1.16 colony, and will provide screenshots today or tommorow.

Seba244c committed:
Fixed typo

Seba244c commented:
Oh wait what, oh I didn't know I could just press accept, I went in and fixed it myself. @ravenbuilder934 can you please cancel your request. I fixed it myself. Sorry.

ravenbuilder934 reviewed (approved):
kk, thanks

Seba244c commented:
Video of PR working
[https://gyazo.com/58b4bf1a136d6b09d6bd0b3057f8f722](https://gyazo.com/58b4bf1a136d6b09d6bd0b3057f8f722)
